types/seat/wlr_seat.c

```diff
@@ -109,33 +115,49 @@ static void seat_handle_bind(struct wl_client *client, void *_wlr_seat,
 	assert(client && wlr_seat);
 
 	struct wlr_seat_client *seat_client =
-		calloc(1, sizeof(struct wlr_seat_client));
+		wlr_seat_client_for_wl_client(wlr_seat, client);
+	bool new_seat_client = false;
 	if (seat_client == NULL) {
-		wl_client_post_no_memory(client);
-		return;
+		seat_client = calloc(1, sizeof(struct wlr_seat_client));
+		if (seat_client == NULL) {
+			wl_client_post_no_memory(client);
+			return;
+		}
+
+		new_seat_client = true;
 	}
-	seat_client->wl_resource =
+
+	struct wl_resource *wl_resource =
```

Comment:
Maybe we can move this block before the previous one and remove `new_seat_client`?

Reply:
Good idea.
How can we cleanly undo a wl_resource_create if the calloc fails, though? just destroy it?

Reply:
Yeah, `wl_resource_destroy` should do it.

Reply:
okay, force pushed.